packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -470,6 +514,15 @@ class FuchsiaDevice extends Device {
     covariant FuchsiaApp app, {
     String userIdentifier,
   }) async {
+    if (await isSession) {
+      // Currently there are no way to close a running app in ermine afaik.
+      // so we just restart the launcher to ensure that it is no longer running.
+      if (!await fuchsiaDeviceTools.sessionControl.restart(this)) {
```

Comment:
This is the equivalent to restarting you laptop every time you build. 

Can you remove this logic entirely? A user can close an application by clicking on the close button in ermine.

Reply:
@chaselatta Out of curiosity, is there a `remove` to go along with the `add`? The Flutter CLI expects to be able to stop a Flutter app e.g. when a user types `q` in the terminal.

Reply:
@zanderso That was the problem. I had ro choose between no closing app snd force close by restart.

Reply:
Not right now but we can request that feature.

Reply:
Got it. Yeah, in the meantime I agree with @chaselatta that this should be removed, and until there's a `remove` potentially added to `session_control` the user can just click the close button.

Reply:
> Got it. Yeah, in the meantime I agree with @chaselatta that this should be removed, and until there's a `remove` potentially added to `session_control` the user can just click the close button.

I too think so. Actually my initial implementation had a no-op stopApp but I just wanted to find a way to make restarting apps more roubust.